tests/50federation/40devicelists.pl

```diff
@@ -284,3 +284,132 @@
          )
       });
    };
+
+test "Device list doesn't change if remote server is down",
+   requires => [
+      $main::OUTBOUND_CLIENT,
+      $main::INBOUND_SERVER,
+      $main::HOMESERVER_INFO[0],
+      local_user_fixture,
+      federation_user_id_fixture(),
+      qw( can_upload_e2e_keys )
+   ],
+
+   check => sub {
+      my (
+         $outbound_client,
+         $inbound_server,
+         $local_server_info,
+         $local_user,
+         $outbound_client_user
+      ) = @_;
+
+      my ( $first_keys_query_body, $second_keys_query_body );
+
+      my $client_user_keys = {
+         device_keys => {
+            $outbound_client_user => {
+               CURIOSITY_ROVER => {
+                  user_id => $outbound_client_user,
+                  device_id => "CURIOSITY_ROVER",
+                  algorithms => ["fast", "and broken"],
+                  keys => {
+                     "curve25519:JLAFKJWSCS" => "3C5BFWi2Y8MaVvjM8M22DBmh24PmgR0nPvJOIArzgyI",
+                     "ed25519:JLAFKJWSCS" => "lEuiRJBit0IG6nUf5pUzWTUEsRVVe/HJkoKuEww9ULI"
+                  },
+                  signatures => {
+                     $outbound_client_user => {"ed25519:JLAFKJWSCS" => "dSO80A01XiigH3uBiDVx/EjzaoycHcjq9lfQX0uWsqxl2giMIiSPR8a4d291W1ihKJL/a+myXS367WT6NAIcBA"}
+                  },
+                  unsigned => {
+                     "device_display_name" => "Curiosity Rover"
+                  },
+               },
+            }
+         },
+      };
+
+      Future->needs_all(
+         $inbound_server->await_request_user_keys_query( $outbound_client_user )->then( sub {
+            my ( $req ) = @_;
+            $req->respond_json($client_user_keys);
+            Future->done(1)
+         }),
+
+         matrix_create_room(
+            $local_user,
+            preset => "public_chat",
+            room_version => "5",
+         )->then( sub {
+            my ( $room_id ) = @_;
+            $outbound_client->join_room(
+               server_name => $local_server_info->server_name,
+               room_id     => $room_id,
+               user_id     => $outbound_client_user,
+            )
+         })->then( sub {
+            do_request_json_for( $local_user,
```

Comment:
Can you just add some comments at the steps here please, e.g. here we do "First we successfully request the keys while the remote server is up", etc.